spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/quartz/TracingJobListenerTest.java

```diff
@@ -0,0 +1,135 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.quartz.Job;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobKey;
+import org.quartz.Scheduler;
+import org.quartz.TriggerBuilder;
+import org.quartz.TriggerKey;
+import org.quartz.impl.JobExecutionContextImpl;
+import org.quartz.spi.OperableTrigger;
+import org.quartz.spi.TriggerFiredBundle;
+import zipkin2.Span;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Answers.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+/**
+ * @author Branden Cash
+ */
+public class TracingJobListenerTest {
+
+	private static final JobKey JOB_KEY = new JobKey("ExampleJob");
+	private static final TriggerKey TRIGGER_KEY = new TriggerKey("ExampleTrigger");
+
+	private TracingJobListener listener;
+
+	private JobExecutionContext context;
+
+	private List<Span> spans = new ArrayList<>();
+
+	@Before
+	public void setUp() {
+		Tracing tracing = Tracing.newBuilder()
+				.spanReporter(spans::add)
+				.currentTraceContext(ThreadLocalCurrentTraceContext.newBuilder()
+						.addScopeDecorator(StrictScopeDecorator.create())
+						.build())
+				.build();
+
+		Tracer tracer = tracing.tracer();
+		listener = new TracingJobListener(tracer);
+
+		TriggerFiredBundle triggerFiredBundle = mock(TriggerFiredBundle.class, withSettings().defaultAnswer(RETURNS_DEEP_STUBS));
+		when(triggerFiredBundle.getTrigger()).thenReturn((OperableTrigger) TriggerBuilder.newTrigger()
+				.forJob(JOB_KEY)
+				.withIdentity(TRIGGER_KEY)
+				.build());
+		context = new JobExecutionContextImpl(mock(Scheduler.class),
+				triggerFiredBundle, mock(Job.class));
+	}
+
+	@Test
+	public void getName_When_AllCases_Expect_ToReturnClassName() {
+		assertThat(listener.getName()).isEqualTo(TracingJobListener.class.getName());
+	}
+
+	@Test
+	public void jobExecutionVetoed_When_ContextContainsSpan_Then_ExpectToCreateSpan() {
+		listener.jobToBeExecuted(context);
+		listener.jobExecutionVetoed(context);
+		takeSpan();
+	}
+
+	@Test
+	public void jobExecutionVetoed_When_ContextDoesNotContainSpan_Then_ExpectNoSpanReported() {
+		listener.jobExecutionVetoed(context);
+		requireNoSpan();
+	}
+
+	@Test
+	public void jobWasExecuted_When_ContextContainsSpan_Then_ExpectToCreateSpan() {
+		listener.jobToBeExecuted(context);
+		listener.jobWasExecuted(context, null);
+		takeSpan();
+	}
+
+	@Test
+	public void jobWasExecuted_When_ContextDoesNotContainSpan_Then_ExpectNoSpanReported() {
+		listener.jobWasExecuted(context, null);
+		requireNoSpan();
+	}
+
+	@Test
+	public void jobWasExecuted_When_ContextContainsSpan_Then_ExpectProperSpanName() {
+		listener.jobToBeExecuted(context);
+		listener.jobWasExecuted(context, null);
+		Span span = takeSpan();
+		assertThat(span.name()).isEqualToIgnoringCase(JOB_KEY.toString());
+		assertThat(span.tags().get("Trigger")).isEqualToIgnoringCase(TRIGGER_KEY.toString());
```

Comment:
literally `quartz.trigger` is a better name. mixed case in lookup keys tend to be problematic